src/sinks/datadog/metrics/encoder.rs

```diff
@@ -415,12 +415,23 @@ fn generate_sketch_metadata(
     origin_product_value: u32,
 ) -> Option<ddmetric_proto::Metadata> {
     generate_origin_metadata(maybe_pass_through, maybe_source_type, origin_product_value).map(
-        |origin| ddmetric_proto::Metadata {
-            origin: Some(ddmetric_proto::Origin {
-                origin_product: origin.product().expect("OriginProduct should be set"),
-                origin_category: origin.category().expect("OriginCategory should be set"),
-                origin_service: origin.service().expect("OriginService should be set"),
-            }),
+        |origin| {
+            if origin.product().is_none() | origin.category().is_none() | origin.service().is_none()
+            {
+                warn!(
+                    message = "Generated sketch origin metadata should have each field set.",
+                    product = origin.product(),
+                    category = origin.category(),
+                    service = origin.service()
+                );
+            }
+            ddmetric_proto::Metadata {
+                origin: Some(ddmetric_proto::Origin {
+                    origin_product: origin.product().expect("OriginProduct should be set"),
+                    origin_category: origin.category().expect("OriginCategory should be set"),
+                    origin_service: origin.service().expect("OriginService should be set"),
+                }),
```

Comment:
I think you meant to update this 🙂 
```suggestion
            ddmetric_proto::Metadata {
                origin: Some(ddmetric_proto::Origin {
                    origin_product: origin.product().unwrap_or_default(),
                    origin_category: origin.category().unwrap_or_default(),
                    origin_service: origin.service().unwrap_or_default(),
                }),
```

Reply:
whoops, yes